### PR TITLE
feat: add support for PG JSONB data type

### PR DESCRIPTION
**Pull request by olavloite**
Adds support for the PostgreSQL `jsonb` data type.

### CODE REVIEW

File: proto-google-cloud-spanner-v1/src/main/java/com/google/spanner/v1/TypeAnnotationCode.java [hunk 1]
```diff
@@ -85,6 +87,8 @@ public enum TypeAnnotationCode implements com.google.protobuf.ProtocolMessageEnu
    * <code>PG_NUMERIC = 2;</code>
    */
   public static final int PG_NUMERIC_VALUE = 2;
+  /** <code>PG_JSONB = 3;</code> */
```
Comment (ansh0l) [inline proto-google-cloud-spanner-v1/src/main/java/com/google/spanner/v1/TypeAnnotationCode.java:90]: nit: Looking at the comment above for `PG_NUMERIC`, I'm wondering if we should add more details in comment for `<code>PG_JSONB = 3;</code>`

File: proto-google-cloud-spanner-v1/src/main/java/com/google/spanner/v1/TypeAnnotationCode.java [hunk 2]
```diff
@@ -57,6 +57,8 @@ public enum TypeAnnotationCode implements com.google.protobuf.ProtocolMessageEnu
    * <code>PG_NUMERIC = 2;</code>
    */
   PG_NUMERIC(2),
+  /** <code>PG_JSONB = 3;</code> */
```
Comment (ansh0l) [inline proto-google-cloud-spanner-v1/src/main/java/com/google/spanner/v1/TypeAnnotationCode.java:60]: nit: Looking at the comment above for `PG_NUMERIC`, I'm wondering if we should add more details in comment for `<code>PG_JSONB = 3;</code>`

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITPgJsonbTest.java [hunk 3]
```diff
@@ -0,0 +1,438 @@
...
+        spannerException.getMessage(),
+        spannerException
+            .getMessage()
+            .contains(
+                "Column with_jsonb_pk.id has type PG.JSONB, but is part of the primary key."));
+  }
+
+  @Test
+  public void testPgJsonbInSecondaryIndex() {
+    // JSONB is not allowed as a primary key.
```
Comment (ansh0l) [inline google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITPgJsonbTest.java:152]: nit: `// JSONB is not allowed in index`

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/connection/RandomResultSetGenerator.java [hunk 4]
```diff
@@ -51,9 +51,12 @@ private static Type[] generateTypes(Dialect dialect) {
               .build()
           : Type.newBuilder().setCode(TypeCode.NUMERIC).build(),
       Type.newBuilder().setCode(TypeCode.STRING).build(),
-      Type.newBuilder()
-          .setCode(dialect == Dialect.POSTGRESQL ? TypeCode.STRING : TypeCode.JSON)
-          .build(),
+      dialect == Dialect.POSTGRESQL
```
Comment (ansh0l) [inline google-cloud-spanner/src/test/java/com/google/cloud/spanner/connection/RandomResultSetGenerator.java:54]: Sounds Good

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/GrpcResultSetTest.java [hunk 5]
```diff
@@ -838,4 +857,23 @@ public void getJsonList() {
     assertTrue(resultSet.next());
     assertEquals(jsonList, resultSet.getJsonList(0));
   }
+
+  @Test
+  public void getPgJsonbList() {
+    List<String> jsonList = new ArrayList<>();
+    jsonList.add("{\"color\":\"red\",\"value\":\"#f00\"}");
+    jsonList.add("{\"special\":\"%😃∮πρότερονแผ่นดินฮั่นเสื่อมሰማይᚻᛖ\"}");
```
Comment (ansh0l) [inline google-cloud-spanner/src/test/java/com/google/cloud/spanner/GrpcResultSetTest.java:865]: Sounds Good

**Review by ansh0l — APPROVED** [review]
The PR looks good to me, there are a couple of minor nits, please take a look.

File: google-cloud-spanner/clirr-ignored-differences.xml [hunk 6]
```diff
@@ -81,4 +81,25 @@
     <className>com/google/cloud/spanner/connection/Connection</className>
     <method>com.google.spanner.v1.ResultSetStats analyzeUpdate(com.google.cloud.spanner.Statement, com.google.cloud.spanner.ReadContext$QueryAnalyzeMode)</method>
   </difference>
```
[the remark this answers is not in the document]
Comment (olavloite) [reply google-cloud-spanner/clirr-ignored-differences.xml:83]: Ah, I did not know that. I'm so used to adding these that I run the clirr checks locally and then add what is needed. So yes; I did face errors, but only on my local machine :-)

File: proto-google-cloud-spanner-v1/src/main/java/com/google/spanner/v1/TypeAnnotationCode.java (hunk 1 above)
Comment (olavloite) [reply proto-google-cloud-spanner-v1/src/main/java/com/google/spanner/v1/TypeAnnotationCode.java:90]: That is probably a good idea, but this is in generated code, so this comment comes from the comments on the proto definition.

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITPgJsonbTest.java (hunk 3 above)
Comment (olavloite) [reply google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITPgJsonbTest.java:152]: Thanks, updated

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITPgJsonbTest.java [hunk 7]
```diff
@@ -0,0 +1,440 @@
...
+import org.junit.ClassRule;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.threeten.bp.Duration;
+
+// TODO: Re-enable when jsonb is GA.
+@Ignore("Feature is not yet generally available")
```
Comment (ansh0l) [inline google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITPgJsonbTest.java:60]: Ack

**Pull request merged**